psutil/_psosx.py

```diff
@@ -212,6 +213,84 @@ def disk_partitions(all=False):
 # =====================================================================
 
 
+def get_smc_temperatures(smc_keys, iter_num, iteratable):
+    """ Receives a dictionary of SMC keys, and a number possible keys
+    If keys can be iterated, replaces the expected %d in the key.
+    Returns a list of all smc_get_temperature results higher than 0
+    """
+    ret = list()
+    if iteratable:
+        for idx in range(iter_num):
+            for key, label in smc_keys.items():
+                result = cext.smc_get_temperatrue(key % idx)
+                if int(result) <= 0:
+                    continue
+                ret.append((label % idx, result, None, None))
+    else:
+        for key, label in smc_keys.items():
+            result = cext.smc_get_temperatrue(key)
+            if int(result) <= 0:
+                continue
+            ret.append((label, result, None, None))
+
+    return ret
+
+
+def sensors_temperatures():
+    """Return CPU temperatures
+    including key name and temperature.
+    """
+    ret = collections.defaultdict(list)
+    # Handle CPU cores
+    num_cpus = cext.cpu_count_phys()
+    core_temps = {
+        "TC%dC": "Core%d",
+    }
+    if num_cpus > 0:
+        results = get_smc_temperatures(core_temps, num_cpus + 1, True)
+        ret["CPU"] = ret["CPU"] + results
+
+    # Handle other CPU sensors
+    cpu_temps = {
+        "TC%dF": "CPU TC%dF",
+        "TC%dG": "CPU TC%dG",
+        "TC%dP": "CPU %d Proximity",
+    }
+    num_packages = cext.cpu_count_packages()
+    if num_packages > 0:
```

Comment:
since `cext.cpu_count_packages()` can return `None` you probably want `if num_packages:`